x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/expression/function/scalar/math/ToNumberFunctionProcessor.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.eql.expression.function.scalar.math;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.xpack.eql.EqlIllegalArgumentException;
+import org.elasticsearch.xpack.ql.expression.gen.processor.Processor;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public class ToNumberFunctionProcessor implements Processor {
+
+    public static final String NAME = "num";
+
+    private final Processor value, base;
+
+    public ToNumberFunctionProcessor(Processor value, Processor base) {
+        this.value = value;
+        this.base = base;
+    }
+
+    public ToNumberFunctionProcessor(StreamInput in) throws IOException {
+        value = in.readNamedWriteable(Processor.class);
+        base = in.readNamedWriteable(Processor.class);
+    }
+
+    @Override
+    public final void writeTo(StreamOutput out) throws IOException {
+        out.writeNamedWriteable(value);
+        out.writeNamedWriteable(base);
+    }
+
+    @Override
+    public Object process(Object input) {
+        return doProcess(value.process(input), base.process(input));
+    }
+
+    private static Number parseDecimal(String source) {
+        try {
+            return Integer.valueOf(source);
```

Comment:
just curious, why Integer? should we use Long as a base type? Long.valueOf()?

Reply:
thank you! good catch, I'll update this